torch/distributions/weibull.py

```diff
@@ -0,0 +1,52 @@
+from numbers import Number
+import math
+import torch
+from torch.distributions import constraints
+from torch.distributions.uniform import Uniform
+from torch.distributions.transformed_distribution import TransformedDistribution
+from torch.distributions.transforms import AffineTransform, ExpTransform, PowerTransform
+from torch.distributions.utils import _finfo, broadcast_all
+from torch.distributions.gumbel import euler_constant
+
+
+class Weibull(TransformedDistribution):
+    r"""
+    Samples from a two-parameter Weibull distribution.
+
+    Example:
+
+        >>> m = Weibull(torch.tensor([1.0]), torch.tensor([1.0]))
+        >>> m.sample()  # sample from a Weibull distribution with scale=1, shape=1
+        tensor([ 0.4784])
+
+    Args:
+        scale (float or Tensor): Scale parameter of distribution (lambda).
+        shape (float or Tensor): Shape parameter of distribution (k).
+    """
+    arg_constraints = {'scale': constraints.positive, 'shape': constraints.positive}
+    support = constraints.positive
+
+    def __init__(self, scale, shape, validate_args=None):
+        self.scale, self.shape = broadcast_all(scale, shape)
+        finfo = _finfo(self.scale)
+        if isinstance(scale, Number) and isinstance(shape, Number):
+            batch_shape = torch.Size()
+            base_dist = Uniform(finfo.tiny, 1 - finfo.eps)
+        else:
+            batch_shape = self.scale.size()
+            base_dist = Uniform(self.scale.new(self.scale.size()).fill_(finfo.tiny), 1 - finfo.eps)
+        transforms = [ExpTransform().inv, AffineTransform(loc=0, scale=-1),
+                      PowerTransform(exponent=self.shape.reciprocal()), AffineTransform(loc=0, scale=self.scale)]
+        super(Weibull, self).__init__(base_dist, transforms, validate_args=validate_args)
+
+    @property
+    def mean(self):
+        return self.scale * torch.exp(torch.lgamma(1 + self.shape.reciprocal()))
+
+    @property
+    def variance(self):
+        return self.scale.pow(2) * (torch.exp(torch.lgamma(1 + 2 * self.shape.reciprocal())) -
+                                    torch.exp(2 * torch.lgamma(1 + self.shape.reciprocal())))
+
+    def entropy(self):
+        return euler_constant * (1 - self.shape.reciprocal()) + torch.log(self.scale * self.shape.reciprocal()) + 1
```

Comment:
Why not `... + torch.log(self.scale / self.shape)) + ...`?

Reply:
Oh I think @tonyduan misunderstood my comment. I suggested that `1 / self.shape` be changed to `self.shape.reciprocal()` because it's faster.

Reply:
Sorry about that @tonyduan .

Reply:
That's ok, if we cache `self.shape_reciprocal` then this code can remain `torch.log(self.scale * self.shape_reciprocal)`